lte/gateway/c/core/oai/lib/s6a_proxy/s6a_client_api.cpp

```diff
@@ -160,7 +161,12 @@ static void s6a_handle_update_location_ans(
   std::cout << "[INFO] sent itti S6A-LOCATION-UPDATE_ANSWER for IMSI: " << imsi
             << std::endl;
   IMSI_STRING_TO_IMSI64((char*) imsi.c_str(), &message_p->ittiMsgHeader.imsi);
-  send_msg_to_task(&s6a_task_zmq_ctx, TASK_MME_APP, message_p);
+
+  if (rat_type == RAT_NR) {
+    send_msg_to_task(&grpc_service_task_zmq_ctx, TASK_AMF_APP, message_p);
```

Comment:
why grpc_service_task_zmq_ctx? you should use s6a_task_zmq_ctx.

Reply:
Facing with runtime error pushing the answer object to destination task queue s6a_task_zmq_ctx as task id: TASK_AMF_APP is not initialized for the same.
...
 init_task_context(
      TASK_S6A, (task_id_t[]){TASK_MME_APP, TASK_S1AP}, 2, handle_message,
      &s6a_task_zmq_ctx);
...

Reply:
S6a does not talk to TASK_S1AP, you can replace that with TASK_AMF_APP.

Reply:
Ok sure Ulas

Reply:
Added the task TASK_AMF_APP to s6a_task_zmq_ctx as in one of the case S6a is sending message ACTIVATE_MESSAGE to TASK_S1AP